test/python/transpiler/test_reset_after_measure_simplification.py

```diff
@@ -26,12 +26,13 @@ def test_simple(self):
         qc = QuantumCircuit(1, 1)
         qc.measure(0, 0)
         qc.reset(0)
-
-        new_qc = ResetAfterMeasureSimplification()(qc)
+        with self.assertWarns(DeprecationWarning):
+            new_qc = ResetAfterMeasureSimplification()(qc)
```

Comment:
Do you think this class raising a deprecation warning is a desired outcome (I'm not sure why it's raised)? Maybe the stack level of the warning isn't correct?

Reply:
The testing harness will trigger if the warning is raised at any stack level even if it's not user code. This is to let us catch internal usage of deprecated functionality in our dependencies. In this case the `ResetAfterMeasureSimplification` pass only works with `c_if` and `condition`. so the warning is coming from inside the pass.

This does raise a good question about what we want to do with that pass, we probably should convert it to an if statement internally. I don't remember what I was thinking when I opened this 2 weeks ago, probably that I would do it in a followup PR. But it would be better if I do it here instead.

Reply:
When I went to do this I remember why I didn't change it. I decided at the time that leaving the pass using `c_if` for 1.x made the most sense as using an `IfElseOp` instead could have been viewed as a potential change in behavior and wanted to save that for 2.0.